concordia/models.py

```diff
@@ -66,6 +68,41 @@ class Collection(models.Model):
     def __str__(self):
         return self.title
 
+    def copy_images_to_collection(self, url, collection_path):
+        result2 = None
+        try:
+            result = download_async_collection.delay(url)
+            result.ready()
+            result.get()
+            result2 = check_completeness.delay()
+            result2.ready()
+            result2.get()
+        except Exception as e:
+            print(e)
+            pass
+
+        if result2 and not result2.state == 'PENDING':
```

Comment:
This block of code should be a separate method and it should be invoked asynchronously (through chained celery task). The create_assets_from_filesystem method would also need to be invoked asynchronously.